test/functional/forknotify.py

```diff
@@ -41,12 +43,19 @@ def run_test(self):
         self.nodes[1].generate(1)
         self.sync_all()
 
+        # Give bitcoind 10 seconds to write the alert notification
+        timeout = 10.0
+        while timeout > 0:
+            if os.path.exists(self.alert_filename) and os.path.getsize(self.alert_filename):
+                break
+            time.sleep(0.1)
+            timeout -= 0.1
+        else:
+            assert False, "-alertnotify did not warn of up-version blocks"
```

Comment:
No good reason. Equivalent behaviour but you're probably right that `raise AssertionError` is better style here.